src/main/java/org/opensearch/plugin/insights/core/service/TopQueriesService.java

```diff
@@ -370,9 +356,7 @@ public List<SearchQueryRecord> getTopQueriesRecordsFromIndex(final String from,
                 List<SearchQueryRecord> records = reader.read(from, to, id);
                 Predicate<SearchQueryRecord> timeFilter = element -> start.toInstant().toEpochMilli() <= element.getTimestamp()
                     && element.getTimestamp() <= end.toInstant().toEpochMilli();
-                List<SearchQueryRecord> filteredRecords = records.stream()
-                    .filter(checkIfInternal.and(timeFilter))
-                    .collect(Collectors.toList());
+                List<SearchQueryRecord> filteredRecords = records.stream().filter(checkIfInternal.and(timeFilter)).toList();
```

Comment:
Use `.collect(Collectors.toList())` notation for compatibility with 2.x java versions